zarr/storage.py

```diff
@@ -2308,63 +2341,52 @@ def __iter__(self):
             list_blobs_prefix = self.prefix + '/'
         else:
             list_blobs_prefix = None
-        for blob in self.client.list_blobs(self.container, list_blobs_prefix):
+        for blob in self.client.list_blobs(list_blobs_prefix):
             yield self._strip_prefix_from_path(blob.name, self.prefix)
 
     def __len__(self):
         return len(self.keys())
 
     def __contains__(self, key):
         blob_name = self._append_path_to_prefix(key)
-        assert len(blob_name) >= 1
-        if self.client.exists(self.container, blob_name):
-            return True
-        else:
-            return False
+        return self.client.get_blob_client(blob_name).exists()
 
     def listdir(self, path=None):
-        from azure.storage.blob import Blob
         dir_path = normalize_storage_path(self._append_path_to_prefix(path))
         if dir_path:
             dir_path += '/'
-        items = list()
-        for blob in self.client.list_blobs(self.container, prefix=dir_path, delimiter='/'):
-            if type(blob) == Blob:
-                items.append(self._strip_prefix_from_path(blob.name, dir_path))
-            else:
-                items.append(self._strip_prefix_from_path(
-                    blob.name[:blob.name.find('/', len(dir_path))], dir_path))
-        return items
+        items = set()
```

Comment:
@jhamman is there a reason you used a `set()` here? Are we relying on `items.add` below to maybe deduplicate something?

If not, I assume that just using a list from the beginning (as required by the `listdir` return value) will be faster.

Reply:
if `delimiter` is specified to `walk_blobs` there is no need for deduplication (although I can see how the Azure Blob API docs seem ambigous), so should be fine with list instead of set IMO.

Reply:
I think I agree with @shikharsg. It has been quite a while and I honestly don't recall my rational on this one.